scripts/import_from_ena_legacy.py

```diff
@@ -0,0 +1,849 @@
+                    parent_animal_list.append(relationship['target'])
+            elif relationship['type'] == 'derived from':
+                if relationship['target'] != biosample_id:
+                    # try to get all records from the relationships
+                    if relationship['target'] not in CACHED_MATERIAL:
+                        retrieve_biosamples_record(es, es_index_prefix, relationship['target'])
+                    related_material = CACHED_MATERIAL[relationship['target']]
+                    if 'material' in related_material and related_material['confirmed'] \
+                            and related_material['material']['text'] == 'organism':
+                        if biosample_id in CACHED_MATERIAL and not CACHED_MATERIAL[biosample_id]['confirmed']:
+                            material_value = {
+                                'text': 'specimen from organism',
+                                'ontologyTerms': 'http://purl.obolibrary.org/obo/OBI_0001479'
+                            }
+                            tmp = {
+                                'accession': biosample_id,
+                                'confirmed': True,
+                                'material': material_value,
+                                'source': 'derived from organism relationship'
+                            }
+                            CACHED_MATERIAL[biosample_id] = tmp
+                        animal = relationship['target']
+
+    if biosample_id not in CACHED_MATERIAL:
+        CACHED_MATERIAL.setdefault(biosample_id, {})
+        CACHED_MATERIAL[biosample_id]['accession'] = biosample_id
+        CACHED_MATERIAL[biosample_id]['confirmed'] = False
+        CACHED_MATERIAL[biosample_id]['material'] = default_material_value
+
+    es_doc = dict()
+    # record labelled as FAANG does not guarantee to be in ES as it may fail the validation,
+    # but should not be dealt with here
+
+    if 'project' in data['characteristics'] and data['characteristics']['project'][0]['text'] == 'FAANG':
+        pass
+        # return -1
+    es_doc['biosampleId'] = biosample_id
+    es_doc['name'] = data['name']
+    es_doc['material'] = CACHED_MATERIAL[biosample_id]['material']
+
+    is_ebi_record = False
+    if biosample_id[0:5] == 'SAMEA':
+        es_doc['id_number'] = biosample_id[5:]
+        is_ebi_record = True
+    else:
+        match = re.search(r'\d+', biosample_id)
+        if match:
+            es_doc['id_number'] = -int(match.group(0))
+    # In the BioSample's code converting NCBI BioSample records into EBI ones,
+    # XML <Description><Title>text</Title></Description> is saved in field "description title"
+    desc_key = get_field_name(data, 'description', 'description title')
+    if desc_key:
+        es_doc['description'] = data['characteristics'][desc_key][0]['text']
+
+    species_key = get_field_name(data, 'organism', 'Organism')
+
+    # https://www.ncbi.nlm.nih.gov/biosample/docs/attributes/?format=xml
```

Comment:
for multi line comments it's better to use """ as code will look much cleaner.

Reply:
done in 27a573a